src/backend/base/langflow/components/prototypes/notdiamond.py

```diff
@@ -0,0 +1,212 @@
+            display_name="Tradeoff",
+            info="The tradeoff between cost and latency for the router to determine the best LLM for a given query.",
+            advanced=False,
+            options=["quality", "cost", "latency"],
+            value="quality",
+        ),
+        BoolInput(
+            name="hash_content",
+            display_name="Hash Content",
+            info="Whether to hash the content before being sent to the NotDiamond API.",
+            advanced=False,
+            value=False,
+        ),
+    ]
+
+    outputs = [Output(display_name="Output", name="output", method="model_select")]
+
+    def model_select(self) -> dict:
+        api_key = SecretStr(self.api_key).get_secret_value() if self.api_key else None
+        input_value = self.input_value
+        system_message = self.system_message
+        messages = self._format_input(input_value, system_message)
+
+        selected_models = []
+        mapped_selected_models = []
+        for model in self.models:
+            model_name = get_model_name(model)
+
+            if model_name in ND_MODEL_MAPPING:
+                selected_models.append(model)
+                mapped_selected_models.append(ND_MODEL_MAPPING[model_name])
+
+        payload = {
+            "messages": messages,
+            "llm_providers": mapped_selected_models,
+            "hash_content": self.hash_content,
+        }
+
+        if self.tradeoff != "quality":
+            payload["tradeoff"] = self.tradeoff
+
+        if self.preference_id and self.preference_id != "":
+            payload["preference_id"] = self.preference_id
+
+        header = {
+            "Authorization": f"Bearer {api_key}",
+            "accept": "application/json",
+            "content-type": "application/json",
+        }
+
+        response = requests.post(
+            "https://api.notdiamond.ai/v2/modelRouter/modelSelect",
+            json=payload,
+            headers=header,
+            timeout=10,
+        )
+
+        result = response.json()
+
+        if "providers" not in result:
+            return result
+
+        providers = result["providers"]
+
+        if len(providers) == 0:
+            return {"chosen_model": None, "text": "No providers returned from NotDiamond API."}
+
+        nd_result = providers[0]
+
+        chosen_model = None
+        for nd_model, selected_model in zip(mapped_selected_models, selected_models, strict=False):
+            if nd_model["provider"] == nd_result["provider"] and nd_model["model"] == nd_result["model"]:
+                chosen_model = selected_model
+                break
+
+        if chosen_model is None:
+            return {"chosen_model": None, "text": "No model found by NotDiamond API."}
+
+        result = get_chat_result(
+            runnable=chosen_model,
+            input_value=input_value,
+            system_message=system_message,
+        )
+
+        return {"chosen_model": get_model_name(chosen_model), "text": result}
```

Comment:
Hey, @barabasicsongor 

I suggest you make some changes regarding errors the user might encounter. 

I don't know how to deal with the case in which no providers return from NotDiamond. What do you suggest?

```suggestion
    outputs = [Output(display_name="Text", name="text_output", method="model_select")]

    def model_select(self) -> Message:
        api_key = SecretStr(self.api_key).get_secret_value() if self.api_key else None
        input_value = self.input_value
        system_message = self.system_message
        messages = self._format_input(input_value, system_message)

        selected_models = []
        mapped_selected_models = []
        for model in self.models:
            model_name = get_model_name(model)

            if model_name in ND_MODEL_MAPPING:
                selected_models.append(model)
                mapped_selected_models.append(ND_MODEL_MAPPING[model_name])

        payload = {
            "messages": messages,
            "llm_providers": mapped_selected_models,
            "hash_content": self.hash_content,
        }

        if self.tradeoff != "quality":
            payload["tradeoff"] = self.tradeoff

        if self.preference_id and self.preference_id != "":
            payload["preference_id"] = self.preference_id

        header = {
            "Authorization": f"Bearer {api_key}",
            "accept": "application/json",
            "content-type": "application/json",
        }

        response = requests.post(
            "https://api.notdiamond.ai/v2/modelRouter/modelSelect",
            json=payload,
            headers=header,
            timeout=10,
        )

        result = response.json()

        if "providers" not in result:
            msg = "No providers returned from NotDiamond API."
            raise ValueError(msg)

        providers = result["providers"]

        if len(providers) == 0:
            msg = "No providers returned from NotDiamond API."
            raise ValueError(msg)

        nd_result = providers[0]

        chosen_model = None
        for nd_model, selected_model in zip(mapped_selected_models, selected_models, strict=False):
            if nd_model["provider"] == nd_result["provider"] and nd_model["model"] == nd_result["model"]:
                chosen_model = selected_model
                break

        if chosen_model is None:
            msg = "No model found by NotDiamond API."
            raise ValueError(msg)

        return get_chat_result(
            runnable=chosen_model,
            input_value=input_value,
            system_message=system_message,
        )
```

Reply:
@ogabrielluiz good point there! We usually recommend that we just fall back to the first model from the models array of the user, so they still receive an answer. I've implemented this in the new commit.